.codegen/api.go.tmpl

```diff
@@ -12,7 +12,94 @@ import (
+	{{.PascalName}}AndWait(ctx context.Context{{if .Request}}, {{.Request.CamelName}} {{.Request.PascalName}}{{end}}, options ...retries.Option[{{.Wait.Poll.Response.PascalName}}]) (*{{.Wait.Poll.Response.PascalName}}, error)
+
+	{{end}}{{if .Pagination}}
+	{{.Comment "// " 80}}
+	// 
+	// This method is generated by Databricks SDK Code Generator.
+	{{.PascalName}}(ctx context.Context{{if .Request}}, request {{.Request.PascalName}}{{end}}) {{ template "paginated-return-type" . }}
+
+	{{.Comment "// " 80}}
+	// 
+	// This method is generated by Databricks SDK Code Generator.
+	{{.PascalName}}All(ctx context.Context{{if .Request}}, request {{.Request.PascalName}}{{end}}) ([]{{ template "type" .Pagination.Entity }}, error)
+		
+	{{end}}{{if .NamedIdMap}}
+	// {{.NamedIdMap.PascalName}} calls [{{.Service.Name}}API.{{.PascalName}}{{if not .NamedIdMap.Direct}}All{{end -}}] and creates a map of results with [{{.NamedIdMap.Entity.PascalName}}]{{ template "field-path" .NamedIdMap.NamePath }} as key and [{{.NamedIdMap.Entity.PascalName}}]{{ template "field-path" .NamedIdMap.IdPath}} as value.
+	//
+	// Returns an error if there's more than one [{{.NamedIdMap.Entity.PascalName}}] with the same {{ template "field-path" .NamedIdMap.NamePath }}.
+	//
+	// Note: All [{{.NamedIdMap.Entity.PascalName}}] instances are loaded into memory before creating a map.
+	// 
+	// This method is generated by Databricks SDK Code Generator.
+	{{.NamedIdMap.PascalName}}(ctx context.Context{{if .Request}}, request {{.Request.PascalName}}{{end}}) (map[string]{{template "type" .NamedIdMap.Id.Entity}}, error)
+	
+	{{end}}{{if .GetByName}}
+	// GetBy{{range .NamedIdMap.NamePath}}{{.PascalName}}{{end}} calls [{{.Service.Name}}API.{{.NamedIdMap.Entity.PascalName}}{{range .NamedIdMap.NamePath}}{{.PascalName}}{{end}}To{{.NamedIdMap.Id.PascalName}}Map] and returns a single [{{.GetByName.PascalName}}].
+	//
+	// Returns an error if there's more than one [{{.NamedIdMap.Entity.PascalName}}] with the same {{ template "field-path" .NamedIdMap.NamePath }}.
+	//
+	// Note: All [{{.NamedIdMap.Entity.PascalName}}] instances are loaded into memory before returning matching by name.
+	// 
+	// This method is generated by Databricks SDK Code Generator.
+	GetBy{{range .NamedIdMap.NamePath}}{{.PascalName}}{{end}}(ctx context.Context, name string) (*{{template "type" .GetByName}}, error)
+		
+	{{end}}{{if .Shortcut}}
+	{{.Comment "// " 80}}
+	{{.Shortcut.PascalName}}(ctx context.Context{{range .Shortcut.Params}}, {{.CamelName}} {{template "type" .Entity}}{{end}}) {{ template "response-type" . }}
+	
+	{{end}}{{if and .Shortcut $hasWaiter}}
+	{{.Shortcut.PascalName}}AndWait(ctx context.Context{{range .Shortcut.Params}}, {{.CamelName}} {{template "type" .Entity}}{{end}}, options ...retries.Option[{{.Wait.Poll.Response.PascalName}}]) (*{{.Wait.Poll.Response.PascalName}}, error)
+	
+	{{end}}
+{{end -}}
+}
+
 func New{{.PascalName}}(client *client.DatabricksClient) *{{.PascalName}}API {
```

Comment:
Should we also remove these types/constructors from the public interface at some point?

Reply:
@mgyucht why should we?

Reply:
This would correspond to us removing the Impl()/WithImpl() methods from the public interface. We would keep them in the SDK for sure, but they don't need to be public anymore.